src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/ShadowApplicationPlugin.kt

```diff
@@ -129,11 +129,14 @@ public abstract class ShadowApplicationPlugin : Plugin<Project> {
   protected open fun Project.configureShadowJarMainClass() {
     val mainClassName = applicationExtension.mainClass
     tasks.shadowJar.configure { task ->
-      task.inputs.property("mainClassName", mainClassName)
```

Comment:
This will result in the `shadowJar` task not being re-run after `applicationExtension.mainClass` changed.